src/integrant/core.cljc

```diff
@@ -24,16 +24,28 @@
     (ref? v)  (list (:key v))
     (coll? v) (mapcat find-refs v)))
 
+(defn find-derived
+  "Return a seq of all entries in a map, m, where the key is derived from the
+  keyword, k. If there are no matching keys, nil is returned."
+  [m k]
+  (seq (filter #(isa? (key %) k) m)))
+
 (defn dependency-graph
-  "Return a dependency graph of all the refs in a config."
+  "Return a dependency graph of all the refs in a config. Resolve derived
+  dependencies."
   [config]
-  (reduce-kv (fn [g k v] (reduce #(dep/depend %1 k %2) g (find-refs v)))
+  (reduce-kv (fn [g k v]
+               (->>
+                (find-refs v)
+                (mapcat (partial find-derived config))
```

Comment:
Since it is a public function I was concerned that throwing an ambiguous key exception might be unexpected behavior. A user might call this function before ambiguous keys have been checked.
That said I agree that it is unlikely to be a problem.